handler/hash_sign.c

```diff
@@ -159,9 +160,22 @@ WEAK unsigned short handler_hash_sign(buffer_t *buffer, uint8_t p1,
     vars.swap_data.alreadySignedInputs++;
     if (vars.swap_data.alreadySignedInputs >=
         vars.swap_data.totalNumberOfInputs) {
-      vars.swap_data.should_exit = 1;
+      if (G_swap_response_ready) {
+        // Safety against trying to make the app sign multiple TX
+        // This code should never be triggered as the app is supposed to exit
+        // after sending the signed transaction
```

Comment:
Nit: `after returning the signature for the last input`, since `hash_sign` is called once for each input.